main/src/ui/unified_window_controller.vala

```diff
@@ -142,6 +142,15 @@ public class UnifiedWindowController : Object {
         }
     }
 
+    public void loop_conversations(bool backwards = false) {
```

Comment:
I think it would be nicer to select the next/previous row based on data available to the UI classes instead of using high-level information on conversations and re-implementing the sorting.

Reply:
That's what I initially thought too, but since they were stored in an unordered data structure and it weren't exposed I chose the other thing.
Your solution looks nice to me though, thanks!